src/core/conversion/dataIndexer.ts

```diff
@@ -100,61 +97,43 @@ export class DataIndexer {
         const timeVal: IVariableValueMeta | undefined = this.getCurrentTimeValue();
         const dataCell: IDataCell = {
             value: this.responseObj.data[this.dataIndex],
-            precision: this.selectedViewMeta[this.contentVariableIndex].values[this.indices[this.contentVariableIndex]].contentComponent?.numberOfDecimals ?? 0,
+            precision: this.selectedViewMeta[this.selectedViewContentVariableIndex].values[this.indices[this.completeViewContentVariableIndex]].contentComponent?.numberOfDecimals ?? 0,
             preliminary: timeVal ? Object.values(timeVal.name)[0].trim().endsWith('*') : false
         };
         if (!dataCell.value) dataCell.missingCode = this.responseObj.missingDataInfo[this.dataIndex];
         return dataCell;
     }
 
-    getRowNames(): TMultiLanguageString[] {
-        let rowNames: TMultiLanguageString[] = [];
-        for (let i = 0; i < this.indices.length; i++) {
-            const variable: IVariableMeta = this.selectedViewMeta[i];
-            const value: IVariableValueMeta = variable.values[this.indices[i]];
-            if ((this.responseObj.rowVariableCodes.includes(variable.code) ||
-                this.responseObj.visualizationSettings.multiselectableVariableCode === variable.code) &&
-                variable.values.length > 1) {
-                rowNames.push(value.name);
-            }
-        }
-        return rowNames;
-    }
-
-    getRowNameGroup(viewSeries: IDataSeries[], rowNames: TMultiLanguageString[]): IDataSeries {
-        const match: IDataSeries | undefined = viewSeries.find(vs => isEqual(vs.rowNameGroup, rowNames));
-        if (match) {
-            return match;
-        }
-        throw new Error("Provided row name group can not be found from the view series");
-    }
-
-    getTargetMap(responseObj: IQueryVisualizationResponse, selectedValueCodes: TVariableSelections): IVariableMeta[] {
-        const targetMap: IVariableMeta[] = responseObj.metaData.map((variable) => {
-            if (responseObj.selectableVariableCodes.includes(variable.code)) {
-                const values: IVariableValueMeta[] = variable.values.filter((value) =>
-                    selectedValueCodes[variable.code].includes(value.code)
-                );
-                if (values.length === 0) {
-                    throw new Error("Provided selected value code can not be found from the metadata");
-                }
-                return { ...variable, values };
+    getSelectedView(responseObj: IQueryVisualizationResponse, selectedValueCodes: TVariableSelections): IVariableMeta[] {
+        const selectableVariables: IVariableMeta[] = responseObj.metaData.filter(v => responseObj.selectableVariableCodes.includes(v.code) || responseObj.visualizationSettings.multiselectableVariableCode == v.code).map((variable) => {
```

Comment:
The multiselectable variable code should always be included in the selectable variable codes list.